### ISSUE TITLE
NGSIv2-NGSIv1 location metadata issues

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3122]
Create an entity E1 using NGSIv2 API with metadata location (old way of marking location in NGSIv1):

```
POST /v2/entities

{
	"id": "E1",
	"type": "T",
	"A": {
		"value": "40, -3",
		"type": "whatever",
		"metadata": {
			"location": {
				"type": "string",
				"value": "WGS84"
			}
		}
	}
}
```

Create an entity E2 using NGSIv1 API with metadata location (old way of marking location in NGSIv1):

```
POST /v1/contextEntities/type/T/id/E2

{
    "attributes": [
        {
            "name": "A",
            "type": "whatever",
            "value": "39,-2",
            "metadatas": [
               {
                  "name": "location",
                  "type": "string",
                  "value": "WGS84"
               }
            ]
        }
    ]
}
```

Getting entities using NGSIv2. Note that metadata location is not rendered in any entity.

```
GET /v2/entities

[
    {
        "id": "E1",
        "type": "T",
        "A": {
            "type": "whatever",
            "value": "40, -3",
            "metadata": {}
        }
    },
    {
        "id": "E2",
        "type": "T",
        "A": {
            "type": "whatever",
            "value": "39,-2",
            "metadata": {}
        }
    }
]
```

Getting entities using NGSIv1. The location metadata is rendered only for E2 (the entity created using NGSIv1):

```
GET /v1/contextEntities

{
    "contextResponses": [
        {
            "contextElement": {
                "type": "T",
                "isPattern": "false",
                "id": "E1",
                "attributes": [
                    {
                        "name": "A",
                        "type": "whatever",
                        "value": "40, -3"
                    }
                ]
            },
            "statusCode": {
                "code": "200",
                "reasonPhrase": "OK"
            }
        },
        {
            "contextElement": {
                "type": "T",
                "isPattern": "false",
                "id": "E2",
                "attributes": [
                    {
                        "name": "A",
                        "type": "whatever",
                        "value": "39,-2",
                        "metadatas": [
                            {
                                "name": "location",
                                "type": "string",
                                "value": "WGS84"
                            }
                        ]
                    }
                ]
            },
            "statusCode": {
                "code": "200",
                "reasonPhrase": "OK"
            }
        }
    ]
}
```

Conclusions:

* The location metadata is ignored in NGSIv2 creation operation at two levels:
   * It cannot be used to mark location. This could be considered functional (althougth it should be at remarked at https://fiware-orion.readthedocs.io/en/master/user/geolocation/index.html documentation) although it could impact in some use cases, e.g entities created/modified by a legacy client using NGSIv1+location_md, the other based on NGSIv2.
   * It is ignored at creation time and not stored as conventional metadata. This could be considered a bug, as even if NGSIv2 ignored the metadata, it should be treated as a regular one.
* Creating a metadata using NGSIv1 which is not shown in the NGSIv2 get operation can be considered as bug. This is probably due to NGSIv1 creating logic doesn't treat it as a regular metadata, so it is not stored in DB.

**Comment by fgalan** [comment]
It seems related to https://github.com/telefonicaid/fiware-orion/issues/3045. Probably the solution for both issues could come together.

**Comment by arigliano** [comment]
Hello, I would like to work on this issue. Can you assign it to me?

**Comment by fgalan** [comment]
Of course. Done!

**Comment by arigliano** [comment]
As far as I understand, the idea would be to:
NGSI v2 Operations:

- **POST /v2/entities**: ignore the location metadata as custom metadata (as for V2 is not the right way to define a  geolocation attribute), and write to DB as normal metadata. So the GET should show it among other regular metadata.
***

NGSI V1:
- **POST /v1/contextEntities/type/T/id/E2** :  use the location metadata to create the location attribute fields (as it is the old V2 way to do this), and in the same time store it as regular metadata, in order to let V2 correctly show in metadata with GET /v2/entities.

**Comment by fgalan** [comment]
Yes, I think your reasoning is correct. In the second case, be careful to not double-rendering the same metadata (one as mark for location and the other retrieved from DB as regular attribute).

Thanks!

**Comment by fgalan** [comment]
Fixed by PR #3346
